### DISCUSSION TITLE
Cascadia Code PL - Some symbols are not displayed

### DISCUSSION THREAD

**Discussion by AdeeelAsif** [JanDeDobbeleer/oh-my-posh#427] [Troubleshoot] [answered]
### Description

Hi everyone, 

some of the symbols are not properly displayed as you can see on the screnshots below. I have the same issue with Powershell 5.1 and Powershell Core 7.0.2

I installed Cascadia Code PL from https://github.com/microsoft/cascadia-code/releases/tag/v2102.03

![image](https://user-images.githubusercontent.com/56124889/108124651-bdbc0f80-70a7-11eb-8e43-2459fb560672.png)

### Environment

- Oh my Posh version: 3.94.2
- Theme: Paradox
- Operating System: Windows 10 
- Shell: Powershell 5.1 (same issue with Powershell core 7.0.2)
- Terminal: Windows Terminal

settings.json content : 

> "list": [
>             {
>                 // Make changes here to the powershell.exe profile.
>                 "guid": "{61c54bbd-c2c6-5271-96e7-009a87ff44bf}",
>                 "name": "Windows PowerShell",
>                 "commandline": "powershell.exe",
>                 "fontFace": "Cascadia Code PL",
>                 "hidden": false
>             },
>             {
>                 "guid": "{574e775e-4f2a-5b96-ac1e-a2962a402336}",
>                 "hidden": false,
>                 "name": "PowerShell",
>                 "fontFace": "Cascadia Code PL",
>                 "source": "Windows.Terminal.PowershellCore"
>             },

### Steps to Reproduce

Import-Module posh-git
Import-Module oh-my-posh
Set-PoshPrompt -Theme paradox

**Comment by lnu** [comment, marked as answer]
Cascadia does not contain all glyphs(same question here https://github.com/JanDeDobbeleer/oh-my-posh/issues/145). From the doc:

> Oh my Posh was designed using [Meslo LGM NF](https://github.com/ryanoasis/nerd-fonts/releases/download/v2.1.0/Meslo.zip), but any Nerd Font should be compatible with the standard themes.

The other option is to replace the missing icons with ones available in cascadia.

**Reply by deathchurch** [reply]
Maybe the author of oh-my-posh should make sure if he is going to make something for PowerShell or indeed Windows Terminal it supports the Microsoft font (Cascadia), or make it more apparent that when upgrading to V3 the install you are left with is terrible. Also make note that Paradox is no longer anywhere to be seen...

Lets be honest most people using windows have probably followed:-

https://www.hanselman.com/blog/how-to-make-a-pretty-prompt-in-windows-terminal-with-powerline-nerd-fonts-cascadia-code-wsl-and-ohmyposh 

So if upgrading to V3 means its not going to look like it was in V2, it should be made clearer in the documentation for upgrading to V3.

**Reply by JanDeDobbeleer** [reply]
@deathchurch ok, that's a pretty arrogant comment. V2 was always created with NF support, as you can see in the [README](https://github.com/JanDeDobbeleer/oh-my-posh2#prerequisites). In fact, CCPL wasn't even around at the time of creation. There's currently one theme that follows the `.minimal` standard meaning it can support Powerline patched fonts that lack a lot of interesting icons allowing a rich and enjoyable terminal experience for a wide range of tools.

**Reply by deathchurch** [reply]
There was no arrogance intended, although i admit the update process was a little bit frustrating.

I still think the documentation could be clearer on the V3 upgrade details, just better mention of what will happen if you are using Cascadia for example. 

From the readme you used as an example, its not entirely correct:-

_From a user perspective, it should give the same experience out-of-the-box, with the added advantage that custom themes are a first class, no code citizen._

This is incorrect it doesn't give the same experience, its one that is suboptimal from V2.

You also mention in another thread you had tried as hard as you could to support Cascadia, i would agree in V2 you had done the hard work and enabled probably the perfect command prompt, as referenced in Hansleman's blog post.

If the docs mentioned the fact Cascadia is not going to cut it really in V3, Paradox theme is no longer there (but referenced a theme that was the same), it would have saved a bit of frustration, and it appears many have had these concerns, as you made note of how you can't believe the same question gets asked about all this.

**Reply by JanDeDobbeleer** [reply]
I'll allow a final explanation to this for future reference:

1. V2 was never created with deliberate support for CCPL, that's a coincidence rather than a design decision. V2 was designed with support for NF, that's a fact (there for sure are themes with unsupported icons too in V2).
2. The installation guide has a [font](https://ohmyposh.dev/docs/installation#font) section, just like V2 did (no regression there, exact same font).
3. Paradox is [available](https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/themes/paradox.omp.json).

If you want to contribute to elevate the experience, we do accept PR's.

**Reply by AntonSmolkov** [reply]
Paradox currently works incorrect even with the Caskaydia Cove Nerd Font =(
![image](https://user-images.githubusercontent.com/5318028/111277967-169caa80-864a-11eb-9732-c860d3fc75f7.png)

**Reply by JanDeDobbeleer** [reply]
@AntonSmolkov are you sure the font is set correctly? Otherwise we'll adjust that icon because indeed, it should work.

**Reply by AntonSmolkov** [reply]
@JanDeDobbeleer 
Just noticed - It works corrently in windows terminal and vscode, but not in the regular powershell window.

**Reply by JanDeDobbeleer** [reply]
In that case the font isn't set correctly in that terminal.

**Comment by seangwright** [comment]
For anyone wondering what is the actual [Nerd Font](https://www.nerdfonts.com/font-downloads) `fontFace` value would be to replace [Cascadia Mono PL](https://docs.microsoft.com/en-us/windows/terminal/cascadia-code), this is what I have in my `settings.json` for Windows Terminal:

```json
"profiles": {
  "defaults": {
    "fontFace": "CaskaydiaCove Nerd Font Mono",
  }
}
```
